AppleALC/ALCUserClientProvider/ALCUserClientProvider.cpp

```diff
@@ -0,0 +1,68 @@
+//
+//  ALCUserClientProvider.cpp
+//  AppleALC
+//
+//  Created by Nick on 10/14/20.
+//  Copyright © 2020 vit9696. All rights reserved.
+//
+
+#include "ALCUserClientProvider.hpp"
+
+OSDefineMetaClassAndStructors(ALCUserClientProvider, IOService);
+
+bool ALCUserClientProvider::start(IOService* provider) {
+	if (!super::start(provider))
+		return false;
+	
+	mHDACodecDevice = IOService::waitForMatchingService(IOService::nameMatching(kIOHDACodecDevice), 3000);	// Wait for 3s
```

Comment:
The unit here is actually nanoseconds. And maybe make the matching dictionary a separate variable and release it later. A good reference here: https://github.com/acidanthera/VirtualSMC/blob/master/Sensors/SMCBatteryManager/SMCBatteryManager.cpp

Reply:
Done.

Reply:
The timeout for `waitForMatchingService` is still left unchanged, and an ideal one could be much larger. On my laptop it will take more than 20 seconds until `HDACodecDevice` is ready. 

And I still think we need debug log here. User could observe the `ReadyForALCVerbs` property for whether it works. For this property, I suppose it's better to display it right under `ALCUserClientProvider` instead of an `IOHDACodecDevice` like `IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/HDEF@1F,3/AppleHDAController@1F,3/IOHDACodecDevice@1F,3,0`

```
[    1.541869]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    4.587347]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    4.643809]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    4.647990]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    4.652781]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    4.654380]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    4.660337]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    6.503196]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    6.834079]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    7.145262]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    7.788265]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    8.199674]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[    8.463444]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   11.536824]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   11.969849]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   19.302273]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   41.877154]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   42.069347]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   42.878221]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   43.065534]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
[   43.113999]: AppleALC    client: @ HDACodecDevice not found, alc-verb will not work.
```

Reply:
Now we wait for 0.1s on each attempt.